demo/contacts/contact-data-source.mjs

```diff
@@ -0,0 +1,33 @@
+export class ContactDataSource {
+  constructor() {
+    this._delay = 1000;
+    this._allContacts = null;
+    this._loading = false;
+    this._loadedAll = false;
+  }
+  get loadedAll() {
+    return this._loadedAll;
+  }
+  get loading() {
+    return this._loading;
+  }
+  get delay() {
+    return this._delay;
+  }
+  set delay(delayMs) {
+    this._delay = delayMs;
+  }
+  async getContacts(count) {
```

Comment:
Maybe name this getNextContacts or getBatchOfContacts or something?

Reply:
done